pulsar-broker/src/main/java/org/apache/pulsar/broker/service/BrokerService.java

```diff
@@ -2608,7 +2608,7 @@ private boolean isSystemTopic(String topic) {
                 maxTopicsPerNamespace = pulsar.getConfig().getMaxTopicsPerNamespace();
             }
 
-            if (maxTopicsPerNamespace > 0) {
+            if (maxTopicsPerNamespace > 0 && !SystemTopicClient.isSystemTopic(topicName)) {
```

Comment:
I don't think the logic here is sufficient to prevent the counting of system topics in the overall topic count for a namespace. On line 2614, we get _all_ topics, which will include system topics in the count of current topics. I think that we'll need to also filter out topics that qualify as system topics.

Reply:
yeah, moved this to the count logic.